### PR TITLE
Added  readme and dependencies

### LINKED ISSUE #5: Adding requirement.txt and Updating README
I recently took a look at the project and noticed that setting it up locally might be a bit tricky for users due to missing dependencies and unclear documentation. 

- Firstly, I suggest creating a `requirements.txt` file with specified versions of the required packages. This will ensure that everyone who wants to run the project locally will have the exact dependencies needed. Here's the updated requirements.txt:

_while trying it to setup I figure out below pkgs are required for me to install_
```
asyncio
anthropic
fastapi
openai
pydantic
pydub
uvicorn
aiohttp
geopy
uule_grabber
serpapi
replicate
python-multipart
```

Additionally, it would be helpful to update the `README.md` file with clear instructions on how to set up the project locally. This should include steps for installing the necessary dependencies, setting up ffmpeg or avconv, and any other prerequisites. Providing a step-by-step guide will make it much easier for new users to get started with the project.

I would more than happy to contribute with updating the documentation. But as I am getting started with codebase I will require some guidance

### PR DESCRIPTION
**Pull request by uma-shankar-TE**
This fixes https://github.com/brilliantlabsAR/noa-assistant/issues/5

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,69 @@
 # Noa Assistant
+ This repository conatains all AI stuff of Noa that can be used in parallel with\
+ a Authentication server to protect APIs and throttled usage\
+Working features:
+1. Conversational AI
+2. Coversation with Photos
```
Comment (trzy) [inline README.md]: Typo: conversation

File: README.md [hunk 2]
```diff
@@ -1 +1,69 @@
...
+3. Input from Audio
```
Comment (trzy) [inline README.md]: Maybe say "Voice transcription" to be more clear

File: README.md [hunk 3]
```diff
@@ -1 +1,69 @@
...
+4. Websearch
```
Comment (trzy) [inline README.md]: Web search should be two words.

File: README.md [hunk 4]
```diff
@@ -1 +1,69 @@
...
+
+### Setup
+1. copy the `.env.example` file to `.env` and fill in the keys
```
Comment (trzy) [inline README.md]: "copy" -> "Copy"

File: README.md [hunk 5]
```diff
@@ -1 +1,69 @@
...
+Note: for reverse image search a image cdn needed, you can use any api which accepts image and returns a url.\
+format of the api should be like this.\
+Or this can be avoided if not using reverse image search
+```sh
+curl -F'file=@path/to/yourfile.png' -Fexpires=2 example.com
+```
+
+1. Create and activate a python virtual enviroment(optional). eg. from [freecodecamp](https://www.freecodecamp.org/news/how-to-setup-virtual-environments-in-python/)
+
+2. Install [ffmpeg](https://ffmpeg.org/download.html), and make sure its available in PATH variable
```
Comment (trzy) [inline README.md:21]: Does noa-assistant really need this? I thought we transmit the raw bytes without any decoding on the server.

**Review by trzy — CHANGES_REQUESTED** [review]
Just some grammar/spelling nit picks. This is otherwise great and thank you for writing it!

File: README.md (hunk 5 above)
Comment (uma-shankar-TE) [reply README.md:21]: Yup for whisper its an overhead, it allows any file format btw, if we fix one format then it will be not needed

**Comment by trzy** [comment]
Ah yes I forgot the conversion happens locally in OpenAI API!Sent from my iPhoneOn Apr 11, 2024, at 9:42 AM, uma-shankar-TE ***@***.***> wrote:﻿
@uma-shankar-TE commented on this pull request.



In README.md:
> +2. Coversation with Photos
+3. Input from Audio
+4. Websearch
+
+### Setup
+1. copy the `.env.example` file to `.env` and fill in the keys
+Note: for reverse image search a image cdn needed, you can use any api which accepts image and returns a url.\
+format of the api should be like this.\
+Or this can be avoided if not using reverse image search
+```sh
+curl ***@***.***/to/yourfile.png' -Fexpires=2 example.com
+```
+
+1. Create and activate a python virtual enviroment(optional). eg. from [freecodecamp](https://www.freecodecamp.org/news/how-to-setup-virtual-environments-in-python/)
+
+2. Install [ffmpeg](https://ffmpeg.org/download.html), and make sure its available in PATH variable

Yup for whisper its an overhead, it allows any file format btw, if we fix one format then it will be not needed

**Review by trzy — APPROVED** [review]

**Pull request merged**